drivers/sensor/bosch/bmm350/bmm350.c

```diff
@@ -0,0 +1,984 @@
+
+#ifdef BMM350_USE_FIXED_POINT
+	data->mag_comp.cross_axis.cross_x_y =
+		fix_sign(cross_x_y, BMM350_SIGNED_8_BIT) * BMM350_MAG_COMP_COEFF_SCALING;
+	data->mag_comp.cross_axis.cross_y_x =
+		fix_sign(cross_y_x, BMM350_SIGNED_8_BIT) * BMM350_MAG_COMP_COEFF_SCALING;
+	data->mag_comp.cross_axis.cross_z_x =
+		fix_sign(cross_z_x, BMM350_SIGNED_8_BIT) * BMM350_MAG_COMP_COEFF_SCALING;
+	data->mag_comp.cross_axis.cross_z_y =
+		fix_sign(cross_z_y, BMM350_SIGNED_8_BIT) * BMM350_MAG_COMP_COEFF_SCALING;
+
+	data->mag_comp.cross_axis.cross_x_y = (data->mag_comp.cross_axis.cross_x_y / 800);
+	data->mag_comp.cross_axis.cross_y_x = (data->mag_comp.cross_axis.cross_y_x / 800);
+	data->mag_comp.cross_axis.cross_z_x = (data->mag_comp.cross_axis.cross_z_x / 800);
+	data->mag_comp.cross_axis.cross_z_y = (data->mag_comp.cross_axis.cross_z_y / 800);
+#endif
+}
+
+static int bmm350_otp_dump_after_boot(const struct device *dev)
+{
+	struct bmm350_data *data = dev->data;
+	int ret = 0;
+	uint16_t otp_word = 0;
+
+	for (uint8_t idx = 0; idx < BMM350_OTP_DATA_LENGTH; idx++) {
+		ret = bmm350_read_otp_word(dev, idx, &otp_word);
+		data->otp_data[idx] = otp_word;
+	}
+
+	data->var_id = (data->otp_data[30] & 0x7f00) >> 9;
+	/* Set the default auto bit reset configuration */
+	data->enable_auto_br = ((data->var_id > BMM350_CURRENT_SHUTTLE_VARIANT_ID) ? BMM350_DISABLE
+										   : BMM350_ENABLE);
+
+	LOG_DBG("bmm350 Find the var id %d\n", data->var_id);
+	/* Update magnetometer offset and sensitivity data. */
+	bmm350_update_mag_off_sens(data);
+
+	if (ret) {
+		LOG_ERR("i2c xfer failed, ret = %d\n", ret);
+	}
+
+	return ret;
+}
+
+/*!
+ * @brief This API gets the PMU command status 0 value
+ */
+static int8_t bmm350_get_pmu_cmd_status_0(struct bmm350_pmu_cmd_status_0 *pmu_cmd_stat_0,
+					  const struct device *dev)
+{
+	/* Variable to store the function result */
+	int8_t ret;
+	uint8_t rx_buf[3] = {0x00};
+
+	if (pmu_cmd_stat_0 != NULL) {
+		/* Get PMU command status 0 data */
+		ret = bmm350_reg_read(dev, BMM350_REG_PMU_CMD_STATUS_0, &rx_buf[0], 3);
+		LOG_DBG("pmu cmd status 0:0x%x\n", rx_buf[2]);
+		if (ret == BMM350_OK) {
+			pmu_cmd_stat_0->pmu_cmd_busy =
+				BMM350_GET_BITS_POS_0(rx_buf[2], BMM350_PMU_CMD_BUSY);
+			pmu_cmd_stat_0->odr_ovwr = BMM350_GET_BITS(rx_buf[2], BMM350_ODR_OVWR);
+			pmu_cmd_stat_0->avr_ovwr = BMM350_GET_BITS(rx_buf[2], BMM350_AVG_OVWR);
+			pmu_cmd_stat_0->pwr_mode_is_normal =
+				BMM350_GET_BITS(rx_buf[2], BMM350_PWR_MODE_IS_NORMAL);
+			pmu_cmd_stat_0->cmd_is_illegal =
+				BMM350_GET_BITS(rx_buf[2], BMM350_CMD_IS_ILLEGAL);
+			pmu_cmd_stat_0->pmu_cmd_value =
+				BMM350_GET_BITS(rx_buf[2], BMM350_PMU_CMD_VALUE);
+		}
+	} else {
+		ret = BMM350_E_NULL_PTR;
+	}
+	return ret;
+}
+
+/*!
+ * @brief This internal API is used to switch from suspend mode to normal mode or forced mode.
+ */
+static int8_t set_powermode(enum bmm350_power_modes powermode, const struct device *dev)
```

Comment:
nit: usually, I always see the `const struct device *` as the first arg for every function. Just something that seems to not follow a pattern